### PR TITLE
bug fix: login check and auth middleware check required security providers

### LINKED ISSUE #1794: setup_lakefs request checks credentials
Setup lakefs API checks user credentials. This causes an issue when the user already has an invalid cookie.
Expected not to check the user credentials in order to succeed in setup a new lakeFS instance and/or clear user credentials when going into `setup` page.

### PR DESCRIPTION
**Pull request by nopcoder**
Close #1794 

1. bug fix: login handler fail on auth check
2. auth middleware will match required security requirements based on swagger definition per route.

### CODE REVIEW

File: pkg/api/auth_middleware.go [hunk 1]
```diff
@@ -20,54 +23,88 @@ var (
 	ErrAuthenticationFailed    = errors.New("error authenticating request")
 )
 
-func AuthMiddleware(logger logging.Logger, authService auth.Service) func(next http.Handler) http.Handler {
+// extractSecurityRequirements using swagger return an array of security requirements set for the request
```
Comment (arielshaqed) [inline pkg/api/auth_middleware.go:26]:
```suggestion
// extractSecurityRequirements using Swagger returns an array of security requirements set for the request.
```
or
```suggestion
// extractSecurityRequirements uses Swagger to return an array of security requirements set for the request.
```

File: pkg/api/auth_middleware.go [hunk 2]
```diff
@@ -20,54 +23,88 @@ var (
...
-			jwtCookie, _ := r.Cookie(JWTCookieName)
-			if jwtCookie != nil {
-				user, err = userByToken(ctx, logger, authService, jwtCookie.Value)
-				if err != nil {
-					writeError(w, http.StatusUnauthorized, err)
-					return
-				}
+			securityRequirements, err := extractSecurityRequirements(router, r)
+			if err != nil {
+				writeError(w, http.StatusBadRequest, err)
```
Comment (arielshaqed) [inline pkg/api/auth_middleware.go:48]: If this goes to users, is the return from `router.FindRoute` safe to show users (and friendly)?

**Review by arielshaqed — APPROVED** [review]
:100: :+1:

File: pkg/api/auth_middleware.go (hunk 2 above)
Comment (nopcoder) [reply pkg/api/auth_middleware.go:48]: 1. The possibilities: "no matching operation was found" or "method not allowed"
2. At this point this call should not fail - The first middleware that responsible to verify the request does the route lookup first. We have 3 places and this is the second one where we are required to do the same lookup again to get the relevant information related to this call.

**Pull request merged**